lib/lz4.c

```diff
@@ -188,7 +188,11 @@
 /*-************************************
 *  Memory routines
 **************************************/
-#ifdef LZ4_USER_MEMORY_FUNCTIONS
+#if defined(LZ4_STATIC_LINKING_ONLY_DISABLE_MEMORY_ALLOCATION)
```

Comment:
How will users be made aware of this new capability ?
Maybe it should be documented ?

Reply:
Do you expect to provide this documentation as part of this PR ?
Or do you want to make it a separate PR ?

Reply:
I'd like to send it as a separate PR.  Sorry!